onnxruntime/core/providers/cuda/nn/dropout.h

```diff
@@ -6,11 +6,22 @@
 #include "core/providers/cuda/cuda_kernel.h"
 #include "core/providers/cuda/nn/dropout_impl.h"
 #include "core/providers/common.h"
-#include "core/framework/random_seed.h"
 
 namespace onnxruntime {
 namespace cuda {
 
+#if defined(CUDA_VERSION) && CUDA_VERSION >= 11000
+#define ALL_IEEE_FLOAT_TENSOR_TYPES           \
+  { DataTypeImpl::GetTensorType<float>(),     \
+    DataTypeImpl::GetTensorType<double>(),    \
+    DataTypeImpl::GetTensorType<MLFloat16>(), \
+    DataTypeImpl::GetTensorType<BFloat16>() }
```

Comment:
>BFloat1 [](start = 32, length = 7)

i believe BFloat16 is not an IEEE floating point format - might be better to omit the IEEE from the name.
where is ALL_IEEE_FLOAT_DATA_TYPES used?
consider defining a type list - that can be used in MLTypeCallDispatcherFromTypeList and also with BuildKernelDefConstraintsFromTypeList in the kernel registration

Reply:
oh i see ALL_IEEE_FLOAT_DATA_TYPES is used in dropout_grad.cc

---
In reply to: [600841457](https://github.com/microsoft/onnxruntime/pull/7116#discussion_r600841457) [](ancestors = 600841457)

Reply:
I found the macro ALL_IEEE_FLOAT_TENSOR_TYPES and ALL_IEEE_FLOAT_DATA_TYPES are spread across multiple files in the repo... 
I agree with your suggestion, but let's do it in another PR.